templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -687,6 +686,33 @@ CRM.$(function($) {
 {/literal}{/if}{literal}
 
 CRM.$(function($) {
+  onPaymentMethodChange();
+  $("#payment_instrument_id").on("change",function(){
+    onPaymentMethodChange();
+  });
+
+  function onPaymentMethodChange() {
+    var paymentInstrument = $('#payment_instrument_id').val();
+    {/literal}
+      {crmAPI var="checkVal" entity="OptionValue" action="getvalue" version="3" option_group_id="payment_instrument" name="Check" return="value"}
+      {crmAPI var="creditVal" entity="OptionValue" action="getvalue" version="3" option_group_id="payment_instrument" name="Credit Card" return="value"}
+    {literal}
+    if (paymentInstrument == {/literal}{$checkVal}{literal}) {
+      $('tr#checkNumber').show();
+      $('tr#cardType').hide();
+      $('tr#cardNumber').hide();
+    }
+    else if (paymentInstrument == {/literal}{$creditVal}{literal}) {
+      $('tr#cardType').show();
+      $('tr#cardNumber').show();
+      $('tr#checkNumber').hide();
+    }
+    else {
+      $('tr#checkNumber').hide();
+      $('tr#cardType').hide();
+      $('tr#cardNumber').hide();
```

Comment:
jQuery selectors can be combined to turn the above three-liners into one-liners. E.g.

    $('tr#checkNumber, tr#cardType, tr#cardNumber').hide();